lib/OpenQA/WebAPI/Controller/API/V1/JobGroup.pm

```diff
@@ -51,30 +51,51 @@ sub list {
     $self->render(json => \@results);
 }
 
+sub load_properties {
+    my ($self) = @_;
+
+    my %properties;
+    for my $param (qw(name parent_id size_limit_gb keep_logs_in_days keep_important_logs_in_days keep_results_in_days keep_important_results_in_days default_priority sort_order description)) {
+        my $value = $self->param($param);
+        $properties{$param} = $value if defined($value);
+    }
+    return \%properties;
+}
+
 sub create {
-    # TODO
+    my ($self) = @_;
+
+    my $group_name = $self->param('name');
+    return $self->render(json => {error => 'No group name specified'}, status => 400) unless $group_name;
+
+    my $group = $self->db->resultset('JobGroups')->create($self->load_properties);
+    return $self->render(json => {error => 'Unable to create job group with specified properties'}, status => 400) unless $group;
+
+    $self->render(json => {id => $group->id});
 }
 
 sub update {
     my ($self) = @_;
 
     my $group_id = $self->param('group_id');
     my $group    = $self->db->resultset('JobGroups')->find($group_id);
-    return $self->render(json => {error => "Job group $group_id does not exist"}) unless $group;
-
-    my %updates;
-    for my $param (qw(name parent_id size_limit_gb keep_logs_in_days keep_important_logs_in_days keep_results_in_days keep_important_results_in_days default_priority sort_order description)) {
-        my $value = $self->param($param);
-        $updates{$param} = $value if defined($value);
-    }
+    return $self->render(json => {error => "Job group $group_id does not exist"}, status => 400) unless $group;
 
-    my $res = $group->update(\%updates);
+    my $res = $group->update($self->load_properties);
     return $self->render(json => {error => "Specified job group $group_id exist but unable to update, though"}) unless $res;
     $self->render(json => {id => $res->id});
 }
 
 sub delete {
-    # TODO
+    my ($self) = @_;
+
+    my $group_id = $self->param('group_id');
+    my $group    = $self->db->resultset('JobGroups')->find($group_id);
+    return $self->render(json => {error => "Job group $group_id does not exist"}, status => 400) unless $group;
+
+    my $res = $group->delete;
```

Comment:
I can't think of any use case of deleting a job group and leaving the jobs. So only deleting empty groups sounds good, but then we need an API call to list jobs in a given group. I'm not sure we have yet.